### PR TITLE
Add --all flag support to molecule reset command

### PR DESCRIPTION
**Pull request by cidrblock**
## Summary

This PR adds the missing `--all` flag support to the `molecule reset` command, bringing it in line with other molecule commands and extending the shared directory functionality previously introduced for shared-state and shared-inventory.

## Problem Statement

The `molecule reset` command was the only core command lacking `--all` flag support, creating inconsistency in the CLI interface. Additionally, the command did not clean up shared ephemeral directories (introduced for shared-state and shared-inventory), leaving artifacts behind during full resets.

### Per-file Changes:

**src/molecule/command/reset.py** (+37 lines, -5 lines)
- Implemented `--all` flag handling with shared directory cleanup logic
- Added conditional cleanup of shared ephemeral directories when `--all` is used
- Integrated with existing `execute_cmdline_scenarios` workflow

**src/molecule/command/base.py** (+1 line, -1 line)  
- Fixed multi-scenario processing bug (changed `return` to `continue` in `_run_scenarios`)
- Ensures all scenarios are processed when using `--all` flag

**tests/unit/command/test_reset.py** (+513 lines, new file)
- Comprehensive test suite with 9 test cases covering:
  - Basic scenario reset functionality
  - Multi-scenario processing with `--all` flag
  - Exclude functionality integration
  - Shared directory cleanup behavior
  - Edge cases and backward compatibility

## Relation to Shared-State/Shared-Inventory

This change complements the shared-state and shared-inventory features introduced months ago by ensuring the `reset` command properly cleans up shared ephemeral directories. When `--all` is specified, the command now performs a "nuclear" cleanup, removing both per-scenario artifacts and shared directories, providing a complete reset functionality.

## Testing

All existing tests pass, and the new test suite provides comprehensive coverage of the reset command functionality including edge cases and integration with the shared directory system.

## Backward Compatibility

This change is fully backward compatible. Existing `molecule reset [scenario]` usage remains unchanged, with the `--all` flag being purely additive functionality.

### CODE REVIEW

File: src/molecule/click_cfg.py [hunk 1]
```diff
@@ -171,6 +171,16 @@ def all_scenarios(self) -> CliOption:
             default=False,
         )
 
+    @property
+    def all_scenarios_reset(self) -> CliOption:
```
Comment (Qalthos) [inline src/molecule/click_cfg.py:175]: Why does this need a different option? Is it just the help text?

Comment (cidrblock) [reply src/molecule/click_cfg.py:175]: yeah, i wanted to bring attention to the fact the shared files were being deleted, it's an unfortuante this about click, no way to just revie the help in the decorator b/c it complains about duplicates

**Review by Qalthos — APPROVED** [review]
I meant in the help text for `reset` itself, but this works too
